File: software/examples/javascript/ExampleSimple.js
var Tinkerforge = require('tinkerforge');

var HOST = 'localhost';
var PORT = 4223;
var UID = 'XYZ'; // Change XYZ to the UID of your Barometer Bricklet 2.0

var ipcon = new Tinkerforge.IPConnection(); // Create IP connection
var b = new Tinkerforge.BrickletBarometerV2(UID, ipcon); // Create device object

ipcon.connect(HOST, PORT,
    function (error) {
        console.log('Error: ' + error);
    }
); // Connect to brickd
// Don't use device before ipcon is connected

ipcon.on(Tinkerforge.IPConnection.CALLBACK_CONNECTED,
    function (connectReason) {
        // Get current air pressure
        b.getAirPressure(
            function (airPressure) {
                console.log('Air Pressure: ' + airPressure/1000.0 + ' mbar');
            },
            function (error) {
                console.log('Error: ' + error);
            }
        );

        // Get current altitude
        b.getAltitude(
            function (altitude) {
                console.log('Altitude: ' + altitude/1000.0 + ' m');
            },
            function (error) {
                console.log('Error: ' + error);
            }
        );
    }
);

console.log('Press key to exit');
process.stdin.on('data',
    function (data) {
        ipcon.disconnect();
        process.exit(0);
    }
);

File: software/examples/rust/example_simple.rs
use std::{io, error::Error};

use tinkerforge::{ip_connection::IpConnection, 
                  barometer_v2_bricklet::*};


const HOST: &str = "localhost";
const PORT: u16 = 4223;
const UID: &str = "XYZ"; // Change XYZ to the UID of your Barometer Bricklet 2.0.

fn main() -> Result<(), Box<dyn Error>> {
    let ipcon = IpConnection::new(); // Create IP connection.
    let b = BarometerV2Bricklet::new(UID, &ipcon); // Create device object.

    ipcon.connect((HOST, PORT)).recv()??; // Connect to brickd.
    // Don't use device before ipcon is connected.

		// Get current air pressure.
let air_pressure = b.get_air_pressure().recv()?;
		println!("Air Pressure: {} mbar", air_pressure as f32 /1000.0);

		// Get current altitude.
let altitude = b.get_altitude().recv()?;
		println!("Altitude: {} m", altitude as f32 /1000.0);

    println!("Press enter to exit.");
    let mut _input = String::new();
    io::stdin().read_line(&mut _input)?;
    ipcon.disconnect();
    Ok(())
}
